src/layers/core/hexagon-layer/hexagon-aggregator.js

```diff
@@ -33,10 +33,14 @@ export function pointToHexbin({data, radius, getPosition}, viewport) {
   // get hexagon radius in mercator world unit
   const radiusInPixel = getRadiusInPixel(radius, viewport);
 
+  const screenPoints = [];
+
   // add world space coordinates to points
-  const screenPoints = data.map(pt => Object.assign({
-    screenCoord: viewport.projectFlat(getPosition(pt))
-  }, pt));
+  for (let i = 0; i < data.length; i++) {
```

Comment:
Nit: I think it is the `apply` that is the issue, not the `map` call. 

The `map` might actually be faster than the repeated `push`es as it can allocate the return array to the right size from the start.